docs/rfcs/0020-collection-containment-membership.md

```diff
@@ -0,0 +1,165 @@
+
+In Linked Data Platform terms, we want to be able to say (in pseudo-RDF)
+
+```
+<lotr> <contains> <fellowship>
+<lotr> <contains> <two-towers>
+<lotr> <contains> <return>
+
+<lotr> <hasMember> <fellowship>
+<lotr> <hasMember> <two-towers>
+<lotr> <hasMember> <return>
+<lotr> <hasMember> <the-last-ringbearer>
+```
+
+## `containedItems` to indicate containment
+
+When a Storage Collection or a IIIF Collection is requested with the `X-IIIF-CS-Show-Extras` header, it has both a `containedItems` and an `items` property.
+
+`containedItems` conveys the containment relationship, and `items` conveys the membership relationship, just as it does in regular IIIF.
+
+This means that the `items` property of an API Storage Collection or an API IIIF Collection is always the `items` property of its public IIIF representation.
+
+It also means that API clients wishing to allow users to **browse the repository storage** just follow `containedItems` and render them as browseable "folders". This is not the same as navigating the IIIF Collections as you would in a viewer, following membership links.
+
+The `items` property of a Storage Collection is _always_ generated dynamically and has the same references in it as the `containedItems` property.
+
+> [!CAUTION]
+> I am suggesting that `items` has vanilla IIIF reference objects, and `containedItems` has API objects with show-extras properties. So even though for a Storage Collection they have the same list of Manifests and/or Collections, the JSON is different; `containedItems` has fields like `publicId`, but `items` is the public facing vanilla IIIF.
+
+The `items` property of a IIIF Collection is generated dynamically as you add more child resources to it as a container, but is **independently editable**.
+
+For both types of Collection, `containedItems` is dynamically generated from the database only. It is not editable as JSON, only through PUT, POST and DELETE operations that change its contained resources.
+
+For a IIIF Collection, we want two things simultaneously that MAY be at odds.
+
+ * we want the `items` property to reflect whatever we want it to, including externally hosted resources, or IIIF resources from elsewhere in our repository
+ * we need to know the containment parent-child relationship so we can see what's in the repository (e.g., navigate a tree in the portal) and have nice hierarchical paths for stored resources in Collections
+
+For the first three volumes of LOTR at least, the `items` from containment is in agreement with the membership `items`, even though the JSON is different (and their `id` values are different (?) - public vs flat?)
+
+For a IIIF Collection, the `items` property is editable as JSON and can be modified without necessarily conflicting with `containedItems`:
+
+* You can let the platform add items to it as you add more child resources to the Collection.
+* You can save the IIIF Collection without an `items` property at all and it will leave it untouched.
```

Comment:
How would this work:
* Create Collection A (empty, no "items") - just a label.
* Add 10 Manifests with parent of Collection A.
  * Collection A now has 10 "items" and 10 "containedItems"
* GET Collection A and remove all "items" then POST.
  * Collection A now has 0 "items" and 10 "containedItems"
* Add 1 Manifest with parent of Collection A
  * Collection A now has 1 "items" and 11 "containedItems"? OR   * Collection A now has 0 "items" and 11 "containedItems", because it's diverged and you're on your own?